app/src/services/webtoon/episodeService.js

```diff
@@ -14,15 +14,44 @@ class EpisodeService {
     return episodes;
   }
 
-  async getEpisodeDetail(episodeId, userId) {
-    const episode = await this.episodeRepository.getEpisodeDetailById(episodeId, userId);
-    if (!episode) {
-      throw new CustomError("해당 에피소드를 찾을 수 없습니다.", 404);
-    }
-    if (episode.hasRated !== undefined && episode.hasRated !== null) {
-      episode.hasRated = !!episode.hasRated;
+  async openEpisodeDetail(episodeId, userId) {
+    let connection;
+    try {
+      connection = await pool.getConnection();
+
+      await connection.beginTransaction();
+
+      const episode = await this.episodeRepository.getEpisodeDetailById(
+        connection,
+        episodeId,
+        userId
+      );
+
+      if (!episode) {
+        await connection.rollback();
+        throw new CustomError("해당 에피소드를 찾을 수 없습니다.", 404);
+      }
+      const webtoonId = episode.webtoonId;
+      if (!webtoonId) {
+        await connection.rollback();
+        throw new CustomError("에피소드에 연결된 웹툰 정보를 찾을 수 없습니다.", 500);
+      }
+
+      await this.episodeRepository.increaseEpisodeViewCount(connection, episodeId);
+      await this.episodeRepository.increaseWebtoonViewCount(connection, webtoonId);
+
+      if (episode.hasRated != null) {
+        episode.hasRated = Boolean(episode.hasRated);
+      }
+
+      await connection.commit();
+      return episode;
+    } catch (error) {
+      if (connection) await connection.rollback();
+      throw new CustomError("회차 불러오는 중 오류 발생", 500);
+    } finally {
```

Comment:
try 내부에서 던진 throw 도 잡히기 때문에 message와 status코드를 활용하려면 아래처럼 구성돼야할거같아요

```suggestion
    } catch (error) {
      if (connection) await connection.rollback();
      
      if (error instanceof CustomError) {
        throw new CustomError(error.message, error.statusCode); // CustomError 인터페이스를 몰라서 속성명은 틀릴 수 있어요. 문법도...
      }
      throw new CustomError("회차 불러오는 중 오류 발생", 500);
    } finally {
```

Reply:
그러네요. 테스트해보니 catch의 CustomError가 기존 에러를 덮어쓰고 있었네요. 말씀해주신 대로 error instanceof CustomError인 경우는 그대로 처리하고, 그 외만 500으로 래핑하는 방식으로 반영하겠습니다. 감사합니다!